api/src/main/java/org/openmrs/module/muzimafingerPrint/api/MuzimafingerPrintService.java

```diff
@@ -26,9 +26,14 @@
     @Transactional
     boolean updatePatient(String patientWithFingerprint) throws JSONException;
 
+    List<PatientFingerPrintModel> findPatients(String searchInput);
+
     MuzimaFingerprint getFingerprintByPatientUUID(String patientUUID);
 
     List<MuzimaFingerprint> getAll();
 
     MuzimaFingerprint findByUniqueId(String uuid);
+
+    @Transactional
+    PatientFingerPrintModel addFingerprintToPatient(String patientWithFingerprint) throws JSONException;
```

Comment:
Please confirm this with Wyclif. I think the new approach is using the transactional annotation at the DAO layer, not the service layer anymore.